### ISSUE TITLE
submit-queue deploymentconfigs invalid

### ISSUE THREAD

**Issue by stevekuznetsov** [openshift/release#185]
On master:

```
$ oc process -f cluster/ci/config/submit-queue/submit_queue.yaml | oc apply -f -
...
Error from server (Invalid): DeploymentConfig "submit-queue" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app":"mungegithub", "component":"submit-queue", "target-repo":"origin"}: `selector` does not match template `labels`
Error from server: error decoding from json: illegal base64 data at input byte 5
Error from server: error decoding from json: illegal base64 data at input byte 5
```

The generated DC is:

```json
{
    "apiVersion": "v1",
    "kind": "DeploymentConfig",
    "metadata": {
        "labels": {
            "app": "mungegithub",
            "component": "submit-queue"
        },
        "name": "submit-queue"
    },
    "spec": {
        "replicas": 1,
        "revisionHistoryLimit": 2,
        "template": {
            "metadata": {
                "labels": {
                    "app": "mungegithub",
                    "component": "submit-queue",
                    "target-repo": "origin"
                }
            },
            "spec": {
                "containers": [
                    {
                        "command": [
                            "/mungegithub",
                            "--dry-run=false",
                            "--alsologtostderr",
                            "--stderrthreshold=INFO",
                            "--config-path=/etc/munge-config/config"
                        ],
                        "image": "submit-queue",
                        "name": "submit-queue",
                        "ports": [
                            {
                                "containerPort": 8080,
                                "name": "status"
                            }
                        ],
                        "resources": {
                            "limits": {
                                "cpu": "2000m"
                            },
                            "requests": {
                                "cpu": "2000m"
                            }
                        },
                        "volumeMounts": [
                            {
                                "mountPath": "/etc/munge-config",
                                "name": "munge-config"
                            },
                            {
                                "mountPath": "/path-label",
                                "name": "path-label"
                            },
                            {
                                "mountPath": "/etc/hmac-token",
                                "name": "hmac-token"
                            },
                            {
                                "mountPath": "/etc/oauth-token",
                                "name": "oauth-token"
                            },
                            {
                                "mountPath": "/gitrepos",
                                "name": "repo"
                            },
                            {
                                "mountPath": "/cache",
                                "name": "cache-volume"
                            }
                        ]
                    }
                ],
                "volumes": [
                    {
                        "configMap": {
                            "name": "submit-queue"
                        },
                        "name": "munge-config"
                    },
                    {
                        "configMap": {
                            "name": "path-label"
                        },
                        "name": "path-label"
                    },
                    {
                        "name": "hmac-token",
                        "secret": {
                            "secretName": "sq-hmac-token"
                        }
                    },
                    {
                        "name": "oauth-token",
                        "secret": {
                            "secretName": "sq-oauth-token"
                        }
                    },
                    {
                        "emptyDir": {},
                        "name": "repo"
                    },
                    {
                        "name": "cache-volume",
                        "persistentVolumeClaim": {
                            "claimName": "origin-cache"
                        }
                    }
                ]
            }
        },
        "triggers": [
            {
                "type": "ConfigChange"
            },
            {
                "imageChangeParams": {
                    "automatic": true,
                    "containerNames": [
                        "submit-queue"
                    ],
                    "from": {
                        "kind": "ImageStreamTag",
                        "name": "submit-queue:latest"
                    }
                },
                "type": "ImageChange"
            }
        ]
    }
}
```

Nothing has changed since I duped this to create the OpenShift-Ansible deployment, and applying that worked fine before... is this a client regression?

/kind bug
/assign @kargakis 
/cc @smarterclayton

**Comment by kargakis** [comment]
@stevekuznetsov you had updated the manifests w/o making the changes in the cluster (https://github.com/openshift/release/commit/2f2441ae209c163145ebcd2e6a622283b3baebd7#diff-b145b556ad63fff1ad69c134c2ee1d8c). Template label changes are the worst to deal with so they should never be left to the next redeployment.

I had stumbled across this when I redeployed prow a while back but didn't know that it was needed for the submit queue too. Anyway, peace in the ci namespace is restored, mungers are happy again with their new labels.
